features/ui/select/select.tsx

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable @next/next/no-img-element */
+import React, { useState, MouseEventHandler, useEffect, useRef } from "react";
+import classNames from "classnames";
+import styles from "./select.module.scss";
+
+type Option = {
+  id: string;
+  value: string;
+};
+
+type SelectProps = {
+  optionsList: Option[];
+  logoSrc?: string;
+  label?: string;
+  hint?: string;
+  error?: Error;
+  disabled?: boolean;
+};
+
+export function Select({
+  optionsList,
+  logoSrc,
+  label,
+  error,
+  hint,
+  disabled,
+}: SelectProps) {
+  const [toggleSelect, setToggleSelect] = useState(false);
+  const [selectedValue, setSelectedValue] = useState<Option | null>(null);
+
+  const selectRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const handleMouseDown = (e: MouseEvent) => {
+      if (disabled) return;
+      if (selectRef.current && !selectRef.current.contains(e.target as Node)) {
+        setToggleSelect(false);
+      }
+    };
+
+    document.addEventListener("mousedown", handleMouseDown);
+
+    return () => document.removeEventListener("mousedown", handleMouseDown);
+  });
+
```

Comment:
One little trick to improve readability: Pass a named function to the useEffect like

```
useEffect(function closeSelectOnClickOutside() {
   ...
}, []);
```

Also you probably forgot to add the dependency array. This way the effect is run on every render and attaches and removes the event listener